movecassdata.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python
+
+import sys
+from util.cass import get_full_cass_dataset, insert_dataset, initialize_worker
+from util.common import StreamKey, TimeRange, ZULU_TIMESTAMP_FORMAT, zulu_timestamp_to_ntp_time
+from ooi_data.postgres.model import *
+from preload_database.database import create_engine_from_url, create_scoped_session
+engine = create_engine_from_url(None)
+session = create_scoped_session(engine)
+MetadataBase.query = session.query_property()
+
+def convert_zulu_to_ntp(zulu_timestamp):
+    try:
+        return zulu_timestamp_to_ntp_time(zulu_timestamp)
+    except (ValueError) as e:
+        usage(str(e))
+        return None
+
+def get_float(value):
+    try:
+        return float(value)
+    except (ValueError):
+        return None
+
+def get_ntp_time(time_stamp):
+    # If timestamp can be converted to float, it's assumed to be NTP
+    # Otherwise it's assumed to be zulu. If neither it will fail cleanly.
+    float_time_stamp = get_float(time_stamp)
+    return float_time_stamp if float_time_stamp else convert_zulu_to_ntp(time_stamp)
+
+def get_stream_key(which, subsite, node, sensor, method, stream_id):
+    # If stream id is all digits, it's assumed to be a valid stream id
+    # else it's assumed to be a valid stream name
+    if stream_id.isdigit():
+        stream = Stream.query.get(int(stream_id))
+    else:
+        stream = Stream.query.filter(Stream.name == stream_id).first()
+
+    if stream is None:
+        usage(which + " stream id (" + stream_id + ") is invalid: must be existing stream id or name")
+        return None
+
+    stream_key =  StreamKey(subsite, node, sensor, method, stream)
+    if stream_key is None:
+        usage("Invalid values were used for the " + which + " stream key")
+
+    return stream_key
+
+def get_time_range(time_stamps):
+    time_stamp_values = time_stamps.split("|")
+    if len(time_stamp_values) != 2:
+        usage("Timestamp argument incorrect")
+        return None
+
+    # Obtain NTP times from the time stamp values
+    begin_time_stamp = get_ntp_time(time_stamp_values[0])
+    end_time_stamp = get_ntp_time(time_stamp_values[1])
+
+    # Confirm valid NTP times were obtained
+    if begin_time_stamp is None or end_time_stamp is None:
+        return None
+
+    if begin_time_stamp > end_time_stamp:
+        usage("Begin timestamp (" + time_stamp_values[0] + ") must not exceed end timestamp (" +
+              time_stamp_values[1] + ")")
+        return None
+
+    # Timestamp range should be within 1995 and 2040
+    if begin_time_stamp < 2997907200 or end_time_stamp > 4449600000:
+        usage("Timestamps outside reasonable range of 1995(2997907200) to 2040(4449600000): " +
```

Comment:
I mainly put this in as an opportunity for discussion. I don't know if it's even worth having a reasonable date range. Jeff said this would only be run by developers. We might want a wide open range of 0 to 9999999999 as well to select all data.

But that said it is relatively easy to use the datetime module's datetime and timedelta to do derive a Zulu datetime of now plus some measure of weeks, days, hours, minutes and/or seconds to do what you said.

Reply:
I guess ideally I'd want a warning and prompt to confirm if the time range was in the future or too far in the past. But I'm fine with whatever you do here.

Reply:
I decided to drop this check